src/components/magnet/magnet-canvas.tsx

```diff
@@ -241,4 +247,9 @@ export class MagnetCanvas extends BaseComponent<IProps, IState> {
       [`rotating${which}`]: isRotating
     });
   }
+
+  private handleSlideArrowComplete = () => {
+    const {simulation} = this.stores;
+    simulation.setSlideArrowComplete(true);
```

Comment:
Could have kept the arrow start/finish values in `magnet-canvas` state like the `isRotating` variables, but this way is fine with me too.